From 289f7f26883b852bb76915b6dfaf6bb6b2be8200 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Fri, 31 Jul 2015 13:41:35 +0700
Subject: [PATCH] qtmultimedia: fix issue drop frame when play video disable
 qos and set max-lateness to -1 to prevent drop frame in
 qtrenderersink and omxh264dec

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 .../mediaplayer/qgstreamerplayersession.cpp        |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp b/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
index 032498b..165d093 100644
--- a/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
+++ b/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
@@ -561,6 +561,8 @@ void QGstreamerPlayerSession::setVideoRenderer(QObject *videoOutput)
     if (!videoSink)
         videoSink = m_nullVideoSink;
 
+    g_object_set(G_OBJECT(videoSink), "qos", false, NULL);
+    g_object_set(G_OBJECT(videoSink), "max-lateness", (gint64)-1, NULL);
 #ifdef DEBUG_PLAYBIN
     qDebug() << "Set video output:" << videoOutput;
     qDebug() << "Current sink:" << (m_videoSink ? GST_ELEMENT_NAME(m_videoSink) : "") <<  m_videoSink
-- 
1.7.9.5

